pkg/solana/chainreader/batch.go

```diff
@@ -146,7 +146,12 @@ func doMethodBatchCall(ctx context.Context, lggr logger.Logger, client MultipleA
 
 		// HACK: workaround for RouterGetWrappedNative
 		if batchCall.ReadName == ccipconsts.MethodNameRouterGetWrappedNative {
-			*(results[idx].returnVal.(*[]byte)) = solana.WrappedSol.Bytes()
+			returnVal, ok := results[idx].returnVal.(*[]byte)
+			if !ok {
+				results[idx].err = fmt.Errorf("returnVal is unexpected type: %T, expected %T", results[idx].returnVal, &[]byte{})
+				continue
+			}
```

Comment:
To be safe, should add a `returnVal != nil` check as well?

Reply:
Good call, added it in the latest commit